lib/core/src/sdk.rs

```diff
@@ -788,6 +788,26 @@ impl LiquidSdk {
                 };
                 payment_destination = SendDestination::Bolt11 { invoice };
             }
+            Ok(InputType::Bolt12 { offer }) => {
+                let amount_sat = req
+                    .amount_sat
+                    .ok_or_else(|| anyhow!("Expected amount when processing BOLT12 offer"))?;
```

Comment:
I see the `lightning` crate is upgraded, is it possible to parse the offer and validate the network / amount (if not optional, matches offer amount or take offer amount if request is not set)?

Reply:
The `offer.amount()` of the parsed `Offer` only [represents](https://github.com/lightningdevkit/rust-lightning/blob/4322b19e619367b371c3b72657c8618a5b9b255e/lightning/src/offers/offer.rs#L620-L623)
> The minimum amount required for a successful payment of a single item

Do you think it makes sense to default to it, if the user chooses no amount?

IMO this is closer to the "pay to Liquid BIP21" use-case, where an user-amount has to be set either in the BIP21 or separately.

Reply:
Yeah,  you're right